manifests/dispatch_router/hub.pp

```diff
@@ -11,26 +11,23 @@
 #
 # @param broker_port
 #   Port of qpidd broker to connect to
-#
-# @param sasl_mech
-#   SASL mechanism to be used from router to broker
-#
-# @param sasl_username
-#   SASL username to be used from router to broker
-#
-# @param sasl_password
-#   SASL password to be used from router to broker
 class foreman_proxy_content::dispatch_router::hub (
   Optional[String] $hub_addr = undef,
   Stdlib::Port $hub_port = 5646,
   String $broker_addr = undef,
   Stdlib::Port $broker_port = 5671,
-  String $sasl_mech = 'PLAIN',
-  String $sasl_username = 'katello_agent',
-  String $sasl_password = extlib::cache_data('foreman_cache_data', 'qpid_router_sasl_password', extlib::random_password(16)),
 ) {
   include foreman_proxy_content::dispatch_router
 
+  class { 'certs::qpid_router_client':
+    hostname => 'qpid_router_katello_agent',
+  }
```

Comment:
Moving the new `qpid_router_client` class declaration inside the hub and connector allow us to configure the generated client certificates differently. For a hub, we want to generate a set of client certificates with a special CN set to `qpid_router_katello_agent` that matches the ACLs set in puppet-katello (https://github.com/theforeman/puppet-katello/pull/384/files#diff-b39fa0a3b1f98c974d4d7d21d4fb0e2d1292dd442ae605d6063c96b9cd9ed932R1). This allows us to be more specific and identify the difference between when Katello and qpid-dispatch connect to qpid. For example:

```
  connection                   cproc      cpid   mech       auth                                                           connected  idle  msgIn  msgOut
  =========================================================================================================================================================
  qpid.[::1]:5671-[::1]:35360  -                 EXTERNAL   qpid_router_katello_agent@QPID                                 6s         4s       0      0
  qpid.[::1]:5671-[::1]:35364  -                 EXTERNAL   pipe-katello-server-nightly-centos7.wareagle.example.com@QPID  2s         0s       0      0
  qpid.[::1]:5671-[::1]:35366  qpid-stat  26105  ANONYMOUS  anonymous@QPID                                                 0s         0s       1      0
```

If both used a client certificate with the hostname, we would not be able to easily tell them apart nor set the ACLs properly. 

On a connector style deployment, such as a proxy with content, we want to deploy the client certificates with the hostname (https://github.com/theforeman/puppet-foreman_proxy_content/pull/327/files#diff-a447b8870e696e6842730621455173b0e03529b5d108cc645e379a45daf2fc30R13) because in that case you have a qpid-dispatch connecting to another qpid-dispatch.